src/compiler/checker.ts

```diff
@@ -6577,6 +6577,21 @@ namespace ts {
             let container = getThisContainer(node, /* includeArrowFunctions */ true);
             let needToCaptureLexicalThis = false;
 
+

```

Comment:
Should we special case `this` before `super`, or should we consider a more general purpose approach to TDZ that encompasses both `this` before `super` and use before declaration for `let` and `const`? I imagine there may be more cases like this in the future.